tests/functional/Glpi/Asset/Capacity/HasOperatingSystemCapacity.php

```diff
@@ -0,0 +1,377 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace tests\units\Glpi\Asset\Capacity;
+
+use DbTestCase;
+use DisplayPreference;
+use Glpi\Asset\AssetDefinition;
+use Glpi\Asset\Capacity\HasHistoryCapacity;
+use Item_OperatingSystem;
+use Profile;
+use Log;
+
+/**
+ * Tests case for the "HasOperatingSystemCapacity" class, which handle the
+ * operating system capacity for assets composition.
+ *
+ * It should validate the following actions are executed when this
+ * capacity is enabled for an asset:
+ * - The asset is registered in $CFG_GLPI["operatingsystem_types"].
+ * - The "Item_OperatingSystem system" tab is added to the asset
+ * - The "Item_OperatingSystem" search options are added to the asset.
+ *
+ * It should validate the following actions are executed when this
+ * capacity is disabled for an asset:
+ * - The asset is unregistered from $CFG_GLPI["operatingsystem_types"].
+ * - All "Item_OperatingSystem" items linked to the asset are deleted.
+ * - All references to "Item_OperatingSystem" and "OperatingSystem" in the
+ * asset history are deleted.
+ * - All display preferences related to "Item_OperatingSystem" for the asset
+ * are deleted.
+ */
+class HasOperatingSystemCapacity extends DbTestCase
+{
+    // #Override
+    protected function initAssetDefinition(
+        ?string $system_name = null,
+        array $capacities = [],
+        array $profiles = [],
+    ): AssetDefinition {
+        return parent::initAssetDefinition(
+            $system_name,
+            array_merge(
+                // Always enable history capacity because some feature of this
+                // capacity depends on it
+                [HasHistoryCapacity::class],
+                $capacities
+            ),
+        );
+    }
```

Comment:
Unless I miss something, this is only required in the `testHistoryDataDeletion` method. You should manually add this capacity only in that method.

```suggestion
```